### PR TITLE
Allow admins to add curators and reviewers to collections

### LINKED ISSUE #665: Add Curators/Reviewers to Collections
As an admin, I want to be able to assign curators and reviewers to a collection so that they can begin reviewing Learning Objects in that collection.

Notes:
- Delete functionality should be included as well

### PR DESCRIPTION
**Pull request by nrwinner**
This PR addresses issue #665 to allow admins to add and remove privileges from users for specific collections. This is a work in progress currently, and there are a few things that need to be done before this can be reviewed and merged.

1. There is a small bug wherein data isn't being passed upwards through the components correctly, and adding a privilege appears to add it to the wrong collection until refreshing the page :heavy_check_mark: 

2. The service is throwing a 500 Internal Service Error for DELETE requests to the route :heavy_check_mark:

3. There are no end-to-end tests for this functionality :heavy_check_mark:

4. I changed the route used to search users for development purposes, that change should be removed in the routes.ts file (just remove the hardcoded values) :heavy_check_mark: 

I'm not really set on the section headers either and how they behave, design input is welcome.

Needs Cyber4All/CLARK-Gateway#76
Needs Cyber4All/user-service#129

Closes #665

### CODE REVIEW

File: src/app/admin/pages/users/users.component.ts [hunk 1]
```diff
@@ -13,9 +13,13 @@ export class UsersComponent implements OnInit {
   users: User[];
   loading = false;
 
+  showPrivileges: boolean;
+  selectedUser: User;
+
   constructor(private user: UserService, private router: Router) { }
 
   ngOnInit() {
+    this.getUsers('');
```
Comment (gshaw1997) [inline src/app/admin/pages/users/users.component.ts]: Why an empty string here?

File: src/app/core/privilege.service.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        { withCredentials: true, responseType: 'text' }
+      )
+      .pipe(
+        retry(3),
+        catchError(this.handleError)
+      )
+      .toPromise();
+  }
+
+  modifyMembership(abvCollectionName: string, userId: string, role: string) {
```
Comment (gshaw1997) [inline src/app/core/privilege.service.ts]: JSDoc

File: src/app/core/privilege.service.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        { withCredentials: true, responseType: 'text' }
+      )
+      .pipe(
+        retry(3),
+        catchError(this.handleError)
+      )
+      .toPromise();
+  }
+
+  removeMembership(abvCollectionName: string, userId: string) {
```
Comment (gshaw1997) [inline src/app/core/privilege.service.ts]: JSDoc

Comment (nrwinner) [reply src/app/core/privilege.service.ts]: Resolved in 630eebd

File: src/app/core/privilege.service.ts (hunk 2 above)
Comment (nrwinner) [reply src/app/core/privilege.service.ts]: Resolved in 630eebd

File: src/app/admin/pages/users/users.component.ts (hunk 1 above)
Comment (nrwinner) [reply src/app/admin/pages/users/users.component.ts]: Resolved in 889a3b3

**Pull request merged**